src/ngMock/angular-mocks.js

```diff
@@ -2600,7 +2602,17 @@ if (window.jasmine || window.mocha) {
     currentSpec = null;
 
     if (injector) {
-      injector.get('$rootElement').off();
+      var cleanUpElems = [originalRootElement];
+      var $rootElement = injector.get('$rootElement');
+      if ($rootElement !== originalRootElement) cleanUpElems.push($rootElement);
+      cleanUpElems.forEach(function(elem) {
+        // The `$rootElement` might not have been created or
+        // a mocked `$rootElement` might not have the standard methods
+        if (elem && elem.off) elem.off();
+        if (elem && elem.removeData) elem.removeData();
+      });
+      if (window.jQuery) window.jQuery.cleanData(cleanUpElems);
```

Comment:
I copied this over from `testabilityPatch.js`, but I think you are right.

From a quick look, I alos think there might be a leak here (and in `testabilityPatch.js`):
Passing an array of jqLite/jQuery elements won't properly clean up. `cleanData` expects an array of "raw" node elements.

I'll try this out.